### ISSUE TITLE
MigrateFileTask doesn't migrate 3.0-style resampled images

### ISSUE THREAD

**Issue by sminnee** [silverstripe/silverstripe-assets#318]
I've found this on silverstripe.org and initially I thought that this was a bug in #314. But unless we want to extend TagsToShortcode to re-generate missing variants, then the real bug is to avoid having variants to missing in the first place.

I'll share some more details once I rebuild my test environment and get some test cases.

**Comment by ichaber** [comment]
For everyone currently experiencing this issue: One workaround for that is to identify the affected pages, that haven't been published after an upgrade to a later SilverStripe 3 version (> 3.3 I believe) and then simply re-publish them.

In my case the following query was helpful to identify affected pages:

```
SELECT ID, URLSegment, Content from SiteTree 
WHERE Content like '%_resampled/ResizedImage%' 
AND Content REGEXP '(_resampled\/ResizedImage)+[a-zA-Z0-9]+[-]{1}' 
AND Content NOT REGEXP '(_resampled\/ResizedImage)+[a-zA-Z0-9]+[\/]{1}'
AND LastEdited <= 'SS3_3_UPGRADE_DATE';
```

**Comment by chillu** [comment]
@sminnee Isn't this covered by https://github.com/silverstripe/silverstripe-assets/issues/313?

**Comment by maxime-rainville** [comment]
#313 should have fixed this issue. If it hasn't, please reopen.

**Comment by chillu** [comment]
I'm reopening this, we've only fixed the 3.2+ variants (e.g. `resampled/ResizedImageWzYwMCwyNzJd-Future-state-homepage-buttons-V33-cropped.png`), not the 3.1 variants (e.g. `_resampled/ResizedImage600272-Future-state-homepage-buttons-V33-cropped.png`). Those are harder to fix (no delimiters between arguments). But the least we should do is detect them, and tell devs that we can't migrate.

On a high level, currently sites which were created under 3.1 or earlier can have broken images after a full 4.x migration, even if they've upgrade to 3.7 before going to 4.x. And since this will only happen on older page content, it's quite easy to miss. And once you find it, it's hard to debug (as Mojmir just found out).

**Comment by maxime-rainville** [comment]
I've raised a separate issue to track the resurgent part of the problem.
